manifests/init.pp

```diff
@@ -29,14 +34,23 @@
     info('puppet_agent performs no actions on Puppet 4+')
   }
   else {
-    class { '::puppet_agent::prepare': } ->
-    class { '::puppet_agent::install': } ->
-    class { '::puppet_agent::config': } ~>
-    class { '::puppet_agent::service': }
+    if $::osfamily == 'windows' {
+      agent_upgrade { 'Windows Puppet Agent Upgrade':
```

Comment:
This skips everything done to fix up `puppet.conf` and the `mcollective` configs.